api/v1alpha1/nicclusterpolicy_types.go

```diff
@@ -101,16 +102,24 @@ type SecondaryNetworkSpec struct {
 	IpamPlugin *ImageSpec `json:"ipamPlugin,omitempty"`
 }
 
+type ClusterNodeSelectorSpec struct {
+	// NodeSelector is a selector which must be true for the pod to fit on a node.
+	// Selector which must match a node's labels for the pod to be scheduled on that node.
+	ClusterNodeSelector map[string]string `json:"nodeSelector,omitempty"`
+}
+
 // NicClusterPolicySpec defines the desired state of NicClusterPolicy
 type NicClusterPolicySpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	OFEDDriver             *OFEDDriverSpec       `json:"ofedDriver,omitempty"`
-	NVPeerDriver           *NVPeerDriverSpec     `json:"nvPeerDriver,omitempty"`
-	RdmaSharedDevicePlugin *DevicePluginSpec     `json:"rdmaSharedDevicePlugin,omitempty"`
-	SriovDevicePlugin      *DevicePluginSpec     `json:"sriovDevicePlugin,omitempty"`
-	SecondaryNetwork       *SecondaryNetworkSpec `json:"secondaryNetwork,omitempty"`
+	ClusterNodeAffinity     *v1.NodeAffinity `json:"nodeAffinity,omitempty"`
```

Comment:
why do we need both ?

Reply:
also this is already in NicClusterPolicy struct, so perhaps just name it `NodeAffinity` ?

Reply:
We need both `nodeSelector` and `nodeAffinity` as they're both supported by kubernetes

Reply:
and why do we need to support both in the policy ?

I just dont fully understand the problem you are trying to solve hence my (possibly dumb) questions.

Reply:
> @adrianchiris - the thought was to give alex to implement the requested feature in #65

So we dont really need both, but just provide the user with an option to apply different policy to different nodes.
we can use just node affinity IMO as it also provides same functionality as node selector (and more)
https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity

that said, if it offers better user-experience im OK with having both

Reply:
Here is issue we're trying to address: #252. Since nodeSelector requires additional logic to not break current workflows I suggest to split it to several PRs. Let's implement node affinity support first